connectors/src/connectors/zendesk/temporal/workflows.ts

```diff
@@ -321,23 +346,16 @@ export async function zendeskCategorySyncWorkflow({
     currentSyncDateMs,
     brandId,
   });
-  if (!wasCategoryUpdated) {
-    return; // nothing to sync
-  }
-
-  let cursor = null; // cursor involved in the pagination of the API
-  let hasMore = true;
-
-  while (hasMore) {
-    const result = await syncZendeskArticlesBatchActivity({
-      connectorId,
-      categoryId,
-      currentSyncDateMs,
-      forceResync,
-      cursor,
-    });
-    hasMore = result.hasMore || false;
-    cursor = result.afterCursor;
+  if (wasCategoryUpdated) {
+    await runWithPagination((cursor) =>
+      syncZendeskArticleBatchActivity({
```

Comment:
`syncZendeskArticlesBatchActivity`

batch activity that syncs zendesk articles :P